src/main/java/io/tus/java/client/persistence/TusURLPersistentStore.java

```diff
@@ -0,0 +1,182 @@
+package io.tus.java.client.persistence;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+
+import org.apache.commons.io.FileUtils;
+
+import io.tus.java.client.TusURLStore;
+
+public class TusURLPersistentStore implements TusURLStore {
+	
+	private static volatile TusUploadFileSet filesWithFingerprints;
+	private String pathOfPersistentXmlFiles = FileUtils.getTempDirectoryPath() + java.io.File.separator + "tus-client-uploads.xml";
+	
+	
+
+	public TusURLPersistentStore() {
+		
+		super();
+		
+		
+		unmarshall();
+		marshall();
+	}
+
+	public TusURLPersistentStore(String pathOfPersistentXmlFiles) {
+		super();
+		this.pathOfPersistentXmlFiles = pathOfPersistentXmlFiles;
+		unmarshall();
+	}
+
+	public void set(String fingerprint, URL url) {
+		
+		unmarshall();
+		if(filesWithFingerprints.getFiles().containsKey(fingerprint))	{
+			
+			  TusUploadFile uf = filesWithFingerprints.getFiles().get(fingerprint);
+			  if(uf != null)	{
+				  uf.setUrl(url);
+			  }
+			  
+		} else  {
+			
+			TusUploadFile uf = new TusUploadFile(); uf.setUrl(url);
+			filesWithFingerprints.getFiles().put(fingerprint, uf);
+		}
+		marshall();
+		
+	}
+	
+	public void set(String fingerprint, URL url, String fileName) {
+		
+		unmarshall();
+		if(filesWithFingerprints.getFiles().containsKey(fingerprint))	{
+			
+			  TusUploadFile uf = filesWithFingerprints.getFiles().get(fingerprint);
+			  if(uf != null)	{
+				  uf.setUrl(url);
+				  uf.setFileName(fileName); 
+			  }
+			  
+		} else  {
+			
+			TusUploadFile uf = new TusUploadFile(); 
+			uf.setUrl(url);
+			filesWithFingerprints.getFiles().put(fingerprint, uf);
+		}
+		marshall();
+	}
+
+	public URL get(String fingerprint) {
+		
+		unmarshall();
+		TusUploadFile uf = filesWithFingerprints.getFiles().get(fingerprint);
+		if(uf != null)	{
+			 return uf.getUrl();
+		}
+		return null;
+	}
+	
+	public TusUploadFile getTusUploadFile(String fingerprint) {

```

Comment:
I see that you are implementing additional methods for storing and retrieving additional information about the uploads, such as the filename. We can implement such a more capable URL store for tus-js-client as well. I think we should do the same for tus-java-client in its next major release: https://github.com/tus/tus-java-client/issues/78